### PR TITLE
Add Branding Manager, add GitHub token

### PR DESCRIPTION
**Pull request by ks129**
Branding manager move got approved in #dev-contrib

~~Added GH token from env variable `GITHUB_TOKEN`.~~ Branding manager ~~and PEPs~~ use this now.

Moved branding manager from SeasonalBot (Cobra) to Python.

### CODE REVIEW

**Review by Xithrius — APPROVED** [review]
You've done an absolutely great job of seamlessly transferring this feature from [sir-lancebot](<https://github.com/python-discord/sir-lancebot>) to this bot. 

I've reviewed this as thoroughly as I could and I didn't find any conflicts or dead functions/methods/classes.

After the conflicts are resolved, this PR should be good to go.

**Comment by Den4200** [comment]
Could you please resolve these conflicts @ks129?

**Comment by ks129** [comment]
@Den4200 Conflicts resolved.

**Review by MarkKoz — CHANGES_REQUESTED** [review]
I'd like to see `seasons.py`, `branding.py`, and the feature-specific constants all under one sub-package.

File: bot/errors.py [hunk 1]
```diff
@@ -18,3 +18,9 @@ def __init__(self, resource_type: str, resource_id: Hashable):
             f"Cannot operate on {self.type.lower()} `{self.id}`; "
             "it is currently locked and in use by another operation."
         )
+
+
+class BrandingError(Exception):
+    """Exception raised by the BrandingManager cog."""
+
+    pass
```
Comment (Den4200) [inline bot/errors.py:26]: The `pass` here can simply be removed.
```suggestion
```

**Review by Den4200 — DISMISSED** [review]

File: bot/exts/backend/branding/_cog.py [hunk 2]
```diff
@@ -135,12 +135,13 @@ def __init__(self, bot: Bot) -> None:
 
         self.days_since_cycle = itertools.cycle([None])
 
-        should_run = self.bot.loop.run_until_complete(self.branding_configuration.get("daemon_active"))
+        self.daemon = None
+        self._startup_task = self.bot.loop.create_task(self._initial_start_daemon())
```
Comment (MarkKoz) [inline bot/exts/backend/branding/_cog.py:139]: The daemon and startup tasks both need to be cancelled when the cog unloads.

**Review by MarkKoz — CHANGES_REQUESTED** [review]
I think it's a little overkill to define an error module for a single error, but it's honestly fine so don't worry about changing it.

**Review by MarkKoz — APPROVED** [review]

**Pull request merged**